ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsStateStore.java

```diff
@@ -93,6 +95,7 @@ LogicalSchema schema() {
   }
 
   private void awaitRunning() {
+    LOG.debug("Waiting for streams thread to be in RUNNING state");
```

Comment:
This busy loop could ultimately hold up threads on the server, for tables that may not be rebalancing too.. (Thread.yield() is just a hint).  Let's fail fast? Now that we have the standby routing, looping until RUNNING may not be necessary at all

Reply:
Done